lib/enrichment/src/get_enrichment_table_record.rs

```diff
@@ -100,6 +131,65 @@ impl Function for GetEnrichmentTableRecord {
             enrichment_tables: registry.as_readonly(),
         }))
     }
+
+    fn compile_argument(
+        &self,
+        args: &[(&'static str, Option<FunctionArgument>)],
+        ctx: &mut FunctionCompileContext,
+        name: &str,
+        expr: Option<&expression::Expr>,
+    ) -> CompiledArgument {
+        match (name, expr) {
+            ("table", Some(expr)) => {
+                let registry = ctx
+                    .get_external_context_mut::<TableRegistry>()
+                    .ok_or(Box::new(vrl_util::Error::TablesNotLoaded) as Box<dyn DiagnosticError>)?;
+
+                let tables = registry
+                    .table_ids()
+                    .into_iter()
+                    .map(Value::from)
+                    .collect::<Vec<_>>();
+
+                let table = expr
+                    .as_enum("table", tables)?
+                    .try_bytes_utf8_lossy()
+                    .expect("table is not valid utf8")
```

Comment:
That's not the cause for the error here, as `String::from_utf8_lossy` always succeeds. The error occurs when we don't have `Bytes`:

https://github.com/vectordotdev/vector/blob/ffe0b8394014622d60428e6c004850d70d5d40b1/lib/vrl/compiler/src/value/convert.rs#L84-L92